testcases/kernel/security/selinux/selinux.c

```diff
@@ -0,0 +1,55 @@
+#include <errno.h>
+#include <fcntl.h>
+
+#include "test.h"
+#include <selinux/selinux.h>
+
+
+char *TCID = "selinux";
+int TST_TOTAL = 1;
+
+static void cleanup(void);
+static void setup(void);
+
+static int old_enforcing;
+
+static void setup(void)
+{
+	tst_require_root();
+	tst_tmpdir();
+
+	if (!is_selinux_enabled())
+		tst_brkm(TBROK, cleanup, "selinux not enabled\n");
+
+	old_enforcing = security_getenforce();
+
+	int setenforce_failure = security_setenforce(1);
+
+	if (setenforce_failure)
+		tst_brkm(TBROK, cleanup,
+			"unable to enforce selinux: %d", errno);
+
+	TEST_PAUSE;
+}
+
+static void cleanup(void)
+{
+	tst_rmdir();
+	security_setenforce(old_enforcing);
+}
+
+int main(void)
+{
+	setup();
+
+	int fd;
+
+	fd = creat("file", 0600);
+	if (-1 == fd)
+		tst_brkm(TBROK, cleanup, "could not create file: %d", errno);
+	else
+		printf("created file %d\n", fd);
+
```

Comment:
You must not use ```printf()``` in the testcases. We have ```tst_resm(TINFO, ...)``` for printing additional information.

Also I still do not understand what is the purpose of the creat() call.

If you want to write a test you need to think of test assertions first. In this case assertions should be derived from libselinux documentation. Looking at it the ```security_setenforce()``` just opens and writes the $selinux_mnt/enforce file so the easiest test can be:

* get enforce value from security_getenforce()
* inverse it and call security_setenforce()
* check that security_getenforce() returns new value
* set the old enforce value

More interesting testcase may do:

* Check that the setenforce was logged by auditd
* Check that policies are actually enforced when setenforce was set